isis/src/base/objs/NumericalApproximation/unitTest.cpp

```diff
@@ -552,57 +552,102 @@ int main(int argc, char *argv[]) {
     for(double a = 0.0; a < 0.8; a += .3) {
       for(double b = 0.8; b > a; b -= .3) {
         s = 0;
-        s = interp5.RefineExtendedTrap(a, b, s, 10);
+        try {
+          s = interp5.RefineExtendedTrap(a, b, s, 10);
+        } catch(IException &e) {
+          e.print();
+        }
+        
         cout << "(" << a << "," << b << ")\t";
-        if(a == 0) cout << "\t";
-        cout << integral(a, b) << " \t";
-        cout << interp3.TrapezoidalRule(a, b) - integral(a, b) << "\t";
-        if(interp3.TrapezoidalRule(a, b) - integral(a, b) == 0) cout << "\t";
-        cout << interp3.Simpsons3PointRule(a, b) - integral(a, b) << "\t";
-        if(interp3.Simpsons3PointRule(a, b) - integral(a, b) == 0) cout << "\t";
-        cout << interp4.Simpsons4PointRule(a, b) - integral(a, b) << "\t";
-        if(interp4.Simpsons4PointRule(a, b) - integral(a, b) == 0) cout << "\t";
-        cout <<      interp4.BoolesRule(a, b) - integral(a, b) << "\t";
-        if(interp4.BoolesRule(a, b) - integral(a, b) == 0) cout << "\t";
-        cout <<                       s - integral(a, b) << "\t";
-        if(s - integral(a, b) == 0) cout << "\t";
-        cout << interp5.RombergsMethod(a, b) - integral(a, b) << endl;
+        if (a == 0) 
+          cout << "\t";
+        try {
+          cout << integral(a, b) << " \t";
+        } catch(IException &e) {
+          e.print();
+        }
+        
+        try {
+          cout << interp3.TrapezoidalRule(a, b) - integral(a, b) << "\t";
+        } catch(IException &e) {
+          e.print();
+        }
+
+        try {
+          if (interp3.TrapezoidalRule(a, b) - integral(a, b) == 0) 
+            cout << "\t";
+          cout << interp3.Simpsons3PointRule(a, b) - integral(a, b) << "\t";
+        } catch (IException &e) {
+          e.print();
+        }
+        
+        try {
+          if (interp3.Simpsons3PointRule(a, b) - integral(a, b) == 0) 
+            cout << "\t";
+          cout << interp4.Simpsons4PointRule(a, b) - integral(a, b) << "\t";
+        } catch (IException &e) {
+          e.print();
+        }
+        
+        try {
+          if (interp4.Simpsons4PointRule(a, b) - integral(a, b) == 0) 
+            cout << "\t";
+          cout <<      interp4.BoolesRule(a, b) - integral(a, b) << "\t";
+        } catch (IException &e) {
+          e.print();
+        }
+        
+        try {
+          if (interp4.BoolesRule(a, b) - integral(a, b) == 0) 
+            cout << "\t";
+          cout <<                       s - integral(a, b) << "\t";
+        } catch (IException &e) {
+          e.print();
+        }
+        
+        try {
+          // Crashes the Mac Arm without throwing an exception 
+          //if (s - integral(a, b) == 0) 
+          //  cout << "\t";
+          //cout << interp5.RombergsMethod(a, b) - integral(a, b) << endl;
+          cout << endl;
+        } catch (IException &e) {
+          e.print();
+        } catch(...) {
+        }
```

Comment:
I don't like just removing the Romberg-Nev tests especially if it still works for linux. NumericalApproximation also has operating system specific truth data. This might be worth doing instead of this change. 

I guess the bigger question is why is it crashing for arm?

Reply:
When working on this earlier, this took a lot of time to even understand why it was failing, as it tests a large number of things in a table, and some are intentionally meant to throw errors. 

The problem is with the algorithm itself. Memory was allocated with a line as: 

double h[maxits+1];

so it was going on the stack. I guess how it is handled is complier-dependent. I replaced that with an std::vector of desired size. It no longer crashes.

I checked out the the old truth now this passes, with tolerance.